cubic-server/Player.cpp

```diff
@@ -1348,6 +1373,43 @@ void Player::teleport(const Vector3<double> &pos)
     Entity::teleport(pos);
 }
 
+void Player::_respawn()
+{
+    this->sendRespawn({
+        "minecraft:overworld", // Dimension
+        "overworld", // World name
+        0, // Hashed seed
+        this->_gamemode, // Gamemode
+        this->_gamemode, // Previous gamemode
+        0, // Is debug
+        0, // Is flat
+        0, // Copy metadata
+        false // Has death location (need dimension name which we don't have yet)
+    });
+
+    for (auto &player : this->_dim->getPlayers()) {
```

Comment:
Please don't get a reference to the players, as that would not increase the reference count to the shared_ptr and the server might just crash if a player gets deleted while you're working on it.